Exercise_2.java

```diff
@@ -8,45 +10,75 @@
   
         StackNode(int data) 
         { 
-            //Constructor here 
+            //Constructor here
+        	this.data = data;
         } 
     } 
     
 	
     public boolean isEmpty() 
     { 
-        //Write your code here for the condition if stack is empty. 
+        return (root == null) ? true : false;
     } 
   
     public void push(int data) 
     { 
-        //Write code to push data to the stack. 
+        //Write code to push data to the stack.
+    	StackNode sn = new StackNode(data);
+    	
+    	if(root == null) {
+    		root = sn;
+    		System.out.println(data + " pushed to stack");
+    	}else {
+    		StackNode temp = root;
+    		root = sn;
+    		root.next = temp;
+    		System.out.println(data + " pushed to stack");
+    	}
+    	
     } 
   
     public int pop() 
     { 	
-	//If Stack Empty Return 0 and print "Stack Underflow"
-        //Write code to pop the topmost element of stack.
-	//Also return the popped element 
+    	int poppedNode = -1;
+    	if(root == null) {
+    		System.out.println("Stack underflow");
+    	}else {
+    		poppedNode = root.data;
+    		root = root.next;
+    	}
+    	return poppedNode;
     } 
   
     public int peek() 
     { 
         //Write code to just return the topmost element without removing it.
+    	if(root == null) {
+    		System.out.println("Stack underflow");
+    		return -1;
+    	}else {
+    		return root.data;
+    	}
     } 
   
 	//Driver code

```

Comment:
Good Job! Please add comment and time and space complexities for all the operations.